File: lista03W/zad1/Szesciokat.java
public class Szesciokat extends Figura {
  private double bok;
  
  public Szesciokat(double bok) {
    this.bok = bok;
  }
  
  public double getPole() {
    return (3 * Math.sqrt(3) * bok * bok) / 2;
  }
  
  public double getObwod() {
    return 6 * bok;
  }
  
  public String getNazwa() {
    return "Szesciokat foremny";
  }
}

File: lista03W/zad2/szesciokat.cpp
#include "szesciokat.hpp"

Szesciokat::Szesciokat(double b) {
  if(b <= 0)
    throw std::invalid_argument("Bok musi być dodatni.");
  
  bok = b;
}

double Szesciokat::getPole() const {
  return (3 * sqrt(3) * bok * bok) / 2;
}

double Szesciokat::getObwod() const {
  return 6 * bok;
}

std::string Szesciokat::getNazwa() const {
  return "Szesciokat foremny";
}